### ISSUE TITLE
warnings_as_errors is bad

### ISSUE THREAD

**Issue by arcusfelis** [inaka/worker_pool#204]
warnings_as_errors is pretty bad for libraries:

```
    ┌─ _build/default/lib/worker_pool/src/wpool_process.erl:
    │
 19 │  -behaviour(gen_server).
    │   ╰── Warning: the callback gen_server:format_status(_,_) is deprecated; use format_status/1 instead

```

when compiling with 27.

Could be fixed with:

```
{overrides, [
   %% Disable warnings_as_errors
   {override, worker_pool, [{erl_opts, []}]}
]}.
```
in the main project, but still, would be nice for `warnings_as_errors` enabled only when building as not-a-dependency.

**Comment by elbrujohalcon** [comment]
The problem is not `warnings_as_errors`, of course… We need to fix that warning by replacing the function and generating a new OTP27-compatible version of this library.
Since I'm not yet working on OTP27, I'll see if I can get myself some time to install OTP27 in my machine and fix the issue.
Thanks for reporting it.

**Comment by paulo-ferraz-oliveira** [comment]
@elbrujohalcon, I got 27 right here. I'd seen this but was lazy to fix. Pushing in a while, hopefully...

**Comment by paulo-ferraz-oliveira** [comment]
A fix for this is out in 6.2.1 (GitHub). It may land in Hex.pm soon...
